app/controllers/discovered_hosts_controller.rb

```diff
@@ -80,6 +80,7 @@ def update
           delete_discovery_attribute_set(@host.id)
           process_success :success_redirect => host_path(@host), :redirect_xhr => request.xhr?
         else
+          @host.expire_token
```

Comment:
That's interesting, and no, I don't have an answer either. It seems very odd. Given the scenario, I'm :+1: for the patch, but add a comment about *why* so we can revisit it after the Rails 4 upgrade - maybe it'll be fixed ;)